repo2docker/docker.py

```diff
@@ -162,10 +160,42 @@ def inspect_image(self, image):
         image = self._apiclient.inspect_image(image)
         return Image(tags=image["RepoTags"], config=image["Config"])
 
+    @contextmanager
+    def docker_login(self, username, password, registry):
+        # Determine existing DOCKER_CONFIG
+        dc_path = Path(
+            os.environ.get("DOCKER_CONFIG", os.path.expanduser("~/.docker/config.json"))
+        )
+
+        with tempfile.TemporaryDirectory() as d:
+            new_dc_path = Path(d) / "config.json"
+            if dc_path.exists():
+                # If there is an existing DOCKER_CONFIG, copy it to new location so we inherit
+                # whatever configuration the user has already set
+                shutil.copy2(dc_path, new_dc_path)
+
+            env = os.environ.copy()
+            subprocess.check_call(
+                # FIXME: This should be using --password-stdin instead
```

Comment:
It might be easier to manipulate the `config.json` file directly:
```python
json.dumps({"auths":{"registry.host":{"auth":b64encode(f"{user}:{token}".encode()).decode()}}})
```

Reply:
@manics I did consider that, but then we'll have to merge the existing ~/.docker/config file with the new setup carefully. I would rather just outsource it to `docker login` like this instead.